deployers/src/main/java/org/ironjacamar/deployers/common/AbstractResourceAdapterDeployer.java

```diff
@@ -274,12 +275,29 @@ public Deployment activate(Connector connector, Activation activation, ClassLoad
          builder.classLoaderPlugin(classLoaderPlugin);
 
          if (connector.getResourceadapter().getResourceadapterClass() != null)
+         {
+            String bootstrapContextName = null;
+            CloneableBootstrapContext bootstrapContext = null;
+            if (activation != null && activation.getBootstrapContext() != null &&
+                  !activation.getBootstrapContext().trim().equals(""))
+            {
+               bootstrapContextName = activation.getBootstrapContext();
+
+               String bootstrapContextIdentifier = bootstrapContextCoordinator
+                     .createIdentifier(connector.getResourceadapter().getResourceadapterClass(),
+                           connector.getResourceadapter().getConfigProperties(), bootstrapContextName);
+               bootstrapContext = bootstrapContextCoordinator
+                     .createBootstrapContext(bootstrapContextIdentifier, bootstrapContextName);
+            }
+            else
+            {
+               bootstrapContext = bootstrapContextCoordinator.getDefaultBootstrapContext();
+            }
```

Comment:
I think it is better to resolve the bootstrap context identifier here, and create the actual instance inside createResourceAdapter()